python/desc/twinkles/generatePhosimInput.py

```diff
@@ -16,122 +16,166 @@
-        obs_metadata.phoSimMetaData['SIM_NSNAP'] = (1, numpy.dtype(int))
-        obs_metadata.phoSimMetaData['SIM_VISTIME'] = (30, numpy.dtype(float))
+        filename = os.path.join("InstanceCatalogs",
+                                "phosim_input_%s.txt" % (obs_metadata.OpsimMetaData['obsHistID']))
         print('Starting Visit: ',
-              obs_metadata.phoSimMetaData['Opsim_obshistid'][0])
+              obs_metadata.OpsimMetaData['obsHistID'])
 
-        compoundStarDBList = [MsStarObj, BhbStarObj, WdStarObj, RRLyStarObj,
-                              CepheidStarObj]
+        compoundStarDBList = [StarObj, CepheidStarObj]
         compoundGalDBList = [GalaxyBulgeObj, GalaxyDiskObj, GalaxyAgnObj]
-        compoundStarICList = [PhoSimCatalogPoint, PhoSimCatalogPoint,
-                              PhoSimCatalogPoint, PhoSimCatalogPoint,
-                              PhoSimCatalogPoint]
+
+        compoundStarICList = [PhoSimCatalogPoint, PhoSimCatalogPoint]
         compoundGalICList = [PhoSimCatalogSersic2D, PhoSimCatalogSersic2D,
                              TwinklesCatalogZPoint]
 
         snphosim = PhoSimCatalogSN(db_obj=snmodel,
-                                   obs_metadata=obs_metadata,
-                                   column_outputs=['EBV'])
+                                   obs_metadata=obs_metadata)
+
         snphosim.writeSedFile = True
         snphosim.suppressDimSN = True
-        snphosim.prefix = 'spectra_files/'
+        snphosim.sn_sedfile_prefix = 'spectra_files/specFile_'
         while True:
             try:
                 starCat = CompoundInstanceCatalog(compoundStarICList,
                                                   compoundStarDBList,
                                                   obs_metadata=obs_metadata,
                                                   constraint='gmag > 11.',
                                                   compoundDBclass=sprinklerCompound)
+
+                starCat._active_connections += available_connections # append already open fatboy connections
+
+                starCat.phoSimHeaderMap = PhoSimHeaderMap
+
+                print("writing starCat ")
                 starCat.write_catalog(filename, chunk_size=10000)
+
                 galCat = CompoundInstanceCatalog(compoundGalICList,
                                                  compoundGalDBList,
                                                  obs_metadata=obs_metadata,
                                                  # constraint='g_ab > 11.',
                                                  compoundDBclass=sprinklerCompound)
+
+                galCat._active_connections = starCat._active_connections # pass along already open fatboy connections
+
+                print("writing galCat")
+
                 galCat.write_catalog(filename, write_mode='a',
-                                     write_header=False, chunk_size=10000)
```

Comment:
Revert back to this line so that `chunk_size=10000` is used for memory-limited machines.